pysat/tests/test_utils_files.py

```diff
@@ -1,58 +1,129 @@
-        # Note: Can be removed if future instrument that uses delimited
-        # filenames is added to routine end-to-end testing
-        fname = ''.join(('test_{year:4d}_{month:2d}_{day:2d}_{hour:2d}',
-                         '_{minute:2d}_{second:2d}_{version:2s}_r02.cdf'))
-        year = np.ones(6) * 2009
-        month = np.ones(6) * 12
-        day = np.array([12, 15, 17, 19, 22, 24])
-        hour = np.array([8, 10, 6, 18, 3, 23])
-        minute = np.array([8, 10, 6, 18, 3, 59])
-        second = np.array([58, 11, 26, 2, 18, 59])
-        version = np.array(['v1', 'v2', 'r1', 'r3', 'v5', 'a6'])
-        file_list = []
-        for i in range(6):
-            file_list.append(fname.format(year=year[i].astype(int),
-                                          month=month[i].astype(int),
-                                          day=day[i], hour=hour[i],
-                                          minute=minute[i], second=second[i],
-                                          version=version[i]))
-
-        file_dict = futils.parse_delimited_filenames(file_list, fname, '_')
-        assert np.all(file_dict['year'] == year)
-        assert np.all(file_dict['month'] == month)
-        assert np.all(file_dict['day'] == day)
-        assert np.all(file_dict['hour'] == hour)
-        assert np.all(file_dict['minute'] == minute)
-        assert np.all(file_dict['day'] == day)
-        assert np.all(file_dict['version'] == version)
-        assert (file_dict['revision'] is None)
-        assert (file_dict['cycle'] is None)
+        del self.fkwargs, self.file_dict, self.kw_format
+
+    def eval_parse_delimited_filename(self):
+        """Evaluate the output of a `parse_delimited_filename` unit test.
+
+        Returns
+        -------
+        bool
+            True if there is data to evalute, False if data dict is empty
+
+        """
+        # Evaluate the returned data dict
+        if len(self.file_dict.keys()) < 2:
+            return False
+
+        # Extract the test lists
+        if len(self.fkwargs) > 0:
+            test_kwargs = {fkey: [kwargs[fkey] for kwargs in self.fkwargs]
+                           for fkey in self.fkwargs[0].keys()}
+        else:
+            test_kwargs = {}
+
+        # Test each of the returned data keys
+        for fkey in self.file_dict.keys():
+            if fkey in test_kwargs:
+                testing.assert_lists_equal(test_kwargs[fkey],
+                                           self.file_dict[fkey])
+            elif fkey == 'files':
+                assert len(self.fkwargs) == len(self.file_dict[fkey]), \
+                    "unexpected file list length"
+            elif fkey != "format_str":
+                assert self.file_dict[fkey] is None, \
+                    "unused format key has a value"
+
+        return True
+
+    # Note: Can be removed if future instrument that uses delimited
+    # filenames is added to routine end-to-end testing
```

Comment:
I probably originally wrote this note but I think we should drop it now. Testing individual functions, along with end-to-end tests, is a better way to ensure we cover all features.
```suggestion
```